SceneRefAttributeValidator.cs

```diff
@@ -176,6 +176,16 @@ List<ReflectionUtil.AttributedField<SceneRefAttribute>> requiredFields
 
         private static object UpdateRef(SceneRefAttribute attr, Component c, FieldInfo field, object existingValue)
         {
+            if (attr.HasFlags(Flag.Editable))
+            {
+                // We convert the reference to a UnityEngine.Object for null checks.
+                if (existingValue as UnityEngine.Object != null)
```

Comment:
Unfortunately it's not quite that easy as we also need to handle arrays and interfaces. It should be enough to check null for objects and interfaces, and for arrays we'd need to check null plus length > 0

Reply:
Updated and tested to handle arrays, PTAL.

Reply:
Looks like interfaces still needs more work.

Reply:
Ok, fixed and tested for interfaces and arrays both.